classes/migration/upgrade/v3_4_0/I8737_SectionEditorsUniqueIndexUpdate.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I8737_SectionEditorsUniqueIndexUpdate.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2000-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I8737_SectionEditorsUniqueIndexUpdate
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Collection;
+use Illuminate\Database\PostgresConnection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\migration\Migration;
+
+class I8737_SectionEditorsUniqueIndexUpdate extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('subeditor_submission_group', function (Blueprint $table) {
+            $this->dropUniqueIndexKey($table);
+            $table->unique(['context_id', 'assoc_id', 'assoc_type', 'user_id', 'user_group_id'], 'section_editors_pkey');
+        });
+    }
+
+    /**
+     * Reverse the migration.
+     */
+    public function down(): void
+    {
+        Schema::table('subeditor_submission_group', function (Blueprint $table) {
+            $this->dropUniqueIndexKey($table);
```

Comment:
Is there a need to check for the index to exist? I think it should always exist, so Laravel's `dropIndex` should be enough, no?

Reply:
@asmecher I added those checks as upgrade tests were failing for `3.2.1` with `PostgreSQL` with error as 

`
ERROR: Upgrade failed: DB: SQLSTATE[42704]: Undefined object: 7 ERROR:  constraint "section_editors_pkey" of relation "subeditor_submission_group" does not exist (SQL: ALTER TABLE subeditor_submission_group DROP CONSTRAINT section_editors_pkey)
`

see at https://app.travis-ci.com/github/pkp/ojs/jobs/597974527 . 

But it seems that it does not solve it . looking into the possible solutions . 

Also `dropIndex` does not work for PostgreSQL as it need to run DB statement as 
```sql
ALTER TABLE subeditor_submission_group DROP CONSTRAINT section_editors_pkey
```
tested in my local `psql 13+` .

Reply:
I didn't watch the PR, but this sounds like https://github.com/pkp/pkp-lib/issues/7670

https://github.com/pkp/pkp-lib/blob/e13b11a2a799a4627430e5193fd83686bbc0295e/classes/migration/upgrade/v3_4_0/I7167_RemoveDuplicatedUserSettingsAndDeprecatedFields.php#L98-L107

Due to differences on how the table was created (old framework vs Laravel stuff), the unique key might exist as a constraint or an index 😅

Reply:
@jonasraoni yes, this is the same problem happening for the PostgreSQL .

Reply:
I'm probably already covering this with https://github.com/pkp/pkp-lib/issues/3573#issuecomment-1480464388 -- this should standardize the constraints/indexes by recreating them using `$table->unique` during the 3.3.0 to 3.4.0 upgrade. Those haven't been merged yet but hopefully this problem will go away then.